messenger/make.py

```diff
@@ -78,14 +133,19 @@ def build_matlab(static=False):
     cfg = get_config()
     # To deal with spaces, remove quotes now, and add
     # to the full commands themselves.
-    matlab_bin = cfg['matlab_bin'].strip('"')
+    if 'matlab_bin' in cfg and cfg['matlab_bin'] != '.':
+        matlab_bin = cfg['matlab_bin'].strip('"')
+    else:  # attempt to autodetect MATLAB filepath
+        matlab_bin = which_matlab()
+        if matlab_bin is None:
+            raise ValueError("specify 'matlab_bin' in cfg file")
     # Get the extension
-    extcmd = '"' + os.path.join(matlab_bin, "mexext") + '"'
-    extension = subprocess.check_output(extcmd, shell=True)
+    extcmd = esc(os.path.join(matlab_bin, "mexext"))
```

Comment:
This doesn't work on python 3, because matlab_bin is bytes and `"mexext"` is a string. 

I propose: 
     extcmd = esc(os.path.join(matlab_bin.decode('utf-8'), "mexext"))